src/CloudNative.CloudEvents.Amqp/AmqpClientExtensions.cs

```diff
@@ -85,6 +84,17 @@ public static class AmqpClientExtensions
                         cloudEvent[attributeName] = property.Value;
                     }
                 }
+                // Populate the data after the rest of the CloudEvent
+                if (message.BodySection is Data data)
+                {
+                    // TODO: Will data.Binary always be populated? Perhaps it'll be data.Buffer?
+                    formatter.DecodeBinaryModeEventData(data.Binary, cloudEvent);
+                }
+                else if (message.BodySection is object)
+                {
+                    throw new ArgumentException("Binary mode data in AMQP message must be in the application data section");
```

Comment:
Should we include relevant spec url?

Reply:
I don't think so, because it could easily move over time.